src/sentry/models/userorgoption.py

```diff
@@ -0,0 +1,129 @@
+from sentry.db.models import FlexibleForeignKey, Model, sane_repr
+from sentry.db.models.fields import EncryptedPickledObjectField
+from sentry.db.models.manager import BaseManager
+
+
+class UserOrgOptionValue(object):
+    all_deploys = '0'
+    committed_only = '1'
+    none = '2'
+
+
+class UserOrgOptionManager(BaseManager):
+    def __init__(self, *args, **kwargs):
+        super(UserOrgOptionManager, self).__init__(*args, **kwargs)
+        self.__metadata = {}
+
+    def __getstate__(self):
+        d = self.__dict__.copy()
+        # we cant serialize weakrefs
+        d.pop('_UserOrgOptionManager__metadata', None)
+        return d
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self.__metadata = {}
+
+    def get_value(self, user, organization, key):
+        result = self.get_all_values(user, organization)
+        return result.get(key, UserOrgOptionValue.committed_only)
+
+    def unset_value(self, user, organization, key):
+        self.filter(user=user, organization=organization, key=key).delete()
+        if not hasattr(self, '_metadata'):
+            return
+        if organization:
+            metakey = (user.pk, organization.pk)
+        else:
+            metakey = (user.pk, None)
+        if metakey not in self.__metadata:
+            return
+        self.__metadata[metakey].pop(key, None)
+
+    def set_value(self, user, organization, key, value):
+        inst, created = self.get_or_create(
+            user=user,
+            organization=organization,
+            key=key,
+            defaults={
+                'value': value,
+            },
+        )
+        if not created and inst.value != value:
+            inst.update(value=value)
+
+        if organization:
+            metakey = (user.pk, organization.pk)
+        else:
+            metakey = (user.pk, None)
+        if metakey not in self.__metadata:
+            return
+        self.__metadata[metakey][key] = value
+
+    def get_all_values(self, user, organization):
+        if organization:
+            metakey = (user.pk, organization.pk)
+        else:
+            metakey = (user.pk, None)
+        if metakey not in self.__metadata:
+            result = dict(
+                (i.key, i.value) for i in
+                self.filter(
+                    user=user,
+                    organization=organization,
+                )
+            )
+            self.__metadata[metakey] = result
+        return self.__metadata.get(metakey, {})
+
+    def clear_cache(self, **kwargs):
+        self.__metadata = {}
+
+    def contribute_to_class(self, model, name):
+        super(UserOrgOptionManager, self).contribute_to_class(model, name)
+        task_postrun.connect(self.clear_cache)
+        request_finished.connect(self.clear_cache)
+
+
+# TODO(dcramer): the NULL UNIQUE constraint here isnt valid, and instead has to
+# be manually replaced in the database. We should restructure this model.
+class UserOrgOption(Model):
```

Comment:
For consistency, I'd really prefer to use `UserOrganizationOption`. We don't shorten `Organization` anywhere in any of our model names. And all the other things that you named this way.